src/core/state.go

```diff
@@ -893,11 +893,13 @@ func (state *BuildState) WaitForPackage(l, dependent BuildLabel, mode ParseMode)
 }
 
 func (state *BuildState) WaitForBuiltTarget(l, dependent BuildLabel, mode ParseMode) *BuildTarget {
-	if t := state.Graph.Target(l); t != nil {
+	t := state.Graph.Target(l)
+	if t != nil {
```

Comment:
Am I reading this right that the behaviour hasn't actually changed here? You don't appear to be referring to `t` later in the function, so this could be:

```
if t := state.Graph.Target(l); t != nil && t.State().IsBuilt {
  return t
}
```

Could also just change back to what it was before to avoid the no-op change?

Reply:
Oops yeah I changed this to debug log some stuff.